src/carousel/carousel.service.ts

```diff
@@ -95,32 +151,41 @@ export class CarouselService {
     id: number,
     file: Express.Multer.File,
     field: CarouselSlideFileProperty,
+    language: Language,
   ) {
-    let carouselSlide = await this.findOne(id);
+    let carouselSlideTranslation = await this.findOne(id, language);
 
-    const oldFileKey = carouselSlide[field];
+    const oldFileKey = carouselSlideTranslation[field];
     const newFileKey = await this.s3.uploadFile(S3_FOLDER, file);
 
     try {
-      carouselSlide = await this.prisma.carouselSlide.update({
-        where: { id },
-        data: { [field]: newFileKey },
-      });
+      const { slide, ...translations } =
+        await this.prisma.carouselSlideTranslation.update({
+          where: { slideId_language: { slideId: id, language } },
+          data: { [field]: newFileKey },
+          include: { slide: true },
+        });
+      carouselSlideTranslation = { ...slide, ...translations };
     } catch {
       await this.s3.deleteObject(newFileKey);
       throw new BadRequestException('Malformed file upload');
     }
 
     await this.s3.garbageCollectOldFile(newFileKey, oldFileKey);
-    return carouselSlide;
+    return carouselSlideTranslation;
```

Comment:
lets rename this to `translatedCarouselSlide` since `carouselSlideTranslation` makes me thing about it as if it's an object from `CarouselSlideTranslation` table